src/components/calendarInput/index.jsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { DayPicker } from 'react-day-picker';
+import { addDays, addYears, endOfYear, format, startOfMonth } from 'date-fns';
+import { useSearchParams } from 'react-router-dom';
+
+import { ReactComponent as CalendarLogo } from 'assets/calendar.svg';
+
+import { setDate } from 'store/slices/inputsSlice';
+import { DayPickerWrapper } from './styled';
+import MainFiltersInput from '../common';
+import { useModal } from '../../hooks/useModal';
+
+const DATE_FORMAT_PATTERN = 'iii d MMM';
+
+export default function CalendarInput() {
+  const [isOpen, onOpenClick, onCloseClick] = useModal();
+
+  const { from, to } = useSelector(state => state.inputs.dates);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const today = new Date();
+  const disabledDays = [
+    {
+      from: startOfMonth(today),
+      to: addDays(today, -1),
+    },
+  ];
+
+  function checkSearchFromValidity(searchFrom) {
+    if (searchFrom) {
+      try {
+        const fromDate = new Date(searchFrom);
+        if (fromDate < today) return false;
+        return true;
+      } catch {
+        return false;
+      }
+    }
+    return false;
+  }
+  function checkSearchToValidity(searchTo) {
+    if (searchTo) {
+      try {
+        const toDate = new Date(searchTo);
+        if (toDate > endOfYear(addYears(today, 1))) return false;
```

Comment:
упрощение читаемости
будет легче добавлять код (на потестить или функциональность)
меньше шансов допустить ошибку
```suggestion
        if (toDate > endOfYear(addYears(today, 1))) {
          return false;
        }
```

Reply:
хорошо, буду стараться ставить скобки